sdlib/d/gc/bin.d

```diff
@@ -72,6 +72,10 @@ private:
 		// FIXME: in contract.
 		assert(mutex.isHeld(), "Mutex not held!");
 
+		if (e.supportsFreeSpace) {
+			e.freeSpaceFlags.clearBitAtomic(index);
+		}
```

Comment:
Also, this seems unecessary because you filter out the stopindex == 0 case anyways.

Reply:
Even if we are worried about this situation - I don't think we need to be, contrary to slice coming from TLS, data segment, the stack or whatever, this can only occur in a program doing something invalid on purpose - then this doesn't work anyways, because the segment is going to appear allocated from the perspective of the bin when sitting in a thread cache, yet it is NOT allocated from the perspective of the program.